### PR TITLE
feat: create rollup deployment params config

### PR DESCRIPTION
**Pull request by spsjvc**
This function replaces `createRollupPrepareConfig` (which is now marked as deprecated), and it preserves the same signature apart from the requirement of the parent public client, so we can make config changes based on the parent chain. It replaces all usage across the codebase. The old function is still there for backwards compat.

The reason why it's named `createRollupPrepareDeploymentParamsConfig`: https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/src/generated.ts#L792

Closes FS-545.

### CODE REVIEW

File: src/getDefaultConfirmPeriodBlocks.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { parentChainIsMainnet } from './parentChainIsMainnet';
+
+export function getDefaultConfirmPeriodBlocks(parentChainId: ParentChainId): bigint {
+  const isMainnet = parentChainIsMainnet(parentChainId);
+
+  if (!isMainnet) {
+    return 150n;
+  }
+
+  return 45_818n;
```
Comment (spsjvc) [inline src/getDefaultConfirmPeriodBlocks.ts:11]: for base, just x6 this

File: src/getDefaultSequencerInboxMaxTimeVariation.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  futureBlocks: bigint;
+  delaySeconds: bigint;
+  futureSeconds: bigint;
+};
+
+export function getDefaultSequencerInboxMaxTimeVariation(
+  parentChainId: ParentChainId,
+): SequencerInboxMaxTimeVariation {
+  return {
+    delayBlocks: BigInt(5_760),
```
Comment (spsjvc) [inline src/getDefaultSequencerInboxMaxTimeVariation.ts:14]: 4 days

File: src/getDefaultSequencerInboxMaxTimeVariation.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    futureBlocks: BigInt(48),
+    delaySeconds: BigInt(86_400),
```
Comment (spsjvc) [inline src/getDefaultSequencerInboxMaxTimeVariation.ts:16]: 4 days in seconds, blocks just divide by 12

Comment (spsjvc) [reply src/getDefaultSequencerInboxMaxTimeVariation.ts:16]: for base, multiply blocks by 6

File: src/createRollupPrepareDeploymentParamsConfig.unit.test.ts
Comment (spsjvc) [inline src/createRollupPrepareDeploymentParamsConfig.unit.test.ts:1]: These are the same test cases used in `createRollupPrepareConfig.unit.test.ts`

File: src/createRollupPrepareDeploymentParamsConfig.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  RequiredParams & { chainConfig?: ChainConfig } & OptionalParams
+>;
+
+export function createRollupPrepareDeploymentParamsConfig(
+  client: PublicClient,
+  { chainConfig, ...params }: CreateRollupPrepareDeploymentParamsConfigParams,
+): CreateRollupPrepareDeploymentParamsConfigResult {
+  const parentChainId = validateParentChain(client);
+
+  const defaultsBasedOnParentChain = {
```
Comment (spsjvc) [inline src/createRollupPrepareDeploymentParamsConfig.ts:33]: This is what's different with the new implementation

**Review by chrstph-dvx — APPROVED** [review]

**Pull request merged**